submissions/team_api.py

```diff
@@ -0,0 +1,240 @@
+          'course_id' Is the ID of the course.
+          'item_id' Is the ID of the item (e.g. the block).
+          'team_id' Is the ID of the team.
+          'submitted_by' Is the ID of the submitting user (same as `submitting_user_id`)
+          'attempt_number' is the attempt number this submission represents for this question.
+          'submitted_at' represents the time this submission was submitted, which can be configured, versus the...
+          'created_at' date, which is when the submission is first created.
+          'submission_uuids' Is a list of the UUIDs of each of the individual `Submission` records that is created.
+
+    Raises:
+        TeamSubmissionRequestError: Raised when there are validation errors for the
+            student item or team submission. This can be caused by the student item
+            missing required values, the submission being too long, the
+            attempt_number is negative, or the given submitted_at time is invalid.
+        SubmissionRequestError: Raised when there are validation errors for the underlying
+            student item or submission. This can be caused by the same reason as
+            the TeamSubmissionRequestError
+        TeamSubmissionInternalError: Raised when submission access causes an
+            internal error.
+        TeamSubmissionInternalError: Raised when submission access causes an
+            internal error when creating the underlying submissions.
+
+    Examples:
+        >>>submitting_user_id = "Tim"
+        >>>item_id = "item_1"
+        >>>course_id = "course_1"
+        >>>team_id = "A Team"
+        >>>team_member_ids = ["Alice", "Bob", "Carol", "Tim"]
+        >>>answer = "The answer is 42."
+        >>> )
+        >>> create_submission_for_team(
+                course_id, item_id, team_id, submitting_user_id, team_member_ids, answer, datetime.utcnow, 1
+            )
+        {
+            'team_submission_uuid': 'blah',
+            'item_id': "item_1",
+            'course_id': "course_1",
+            'team_id': "A Team",
+            'submitted_by': "Tim",
+            'attempt_number': 1,
+            'submitted_at': datetime.datetime(2014, 1, 29, 17, 14, 52, 649284 tzinfo=<UTC>),
+            'created_at': datetime.datetime(2014, 1, 29, 17, 14, 52, 668850, tzinfo=<UTC>),
+            'answer': u'The answer is 42.',
+            'submission_uuids': ['alice-uuid', 'bob-uuid', 'carol-uuid', 'tim-uuid'],
+        }
+    """
+
+    # I have no clue what to do with attempt_number. The normal API checks if there are any duplicate submissions and
+    # incrememnts attempt_number, but we prevent duplicate team submissions, so I can't copy that logic flow.
+    # I thought maybe we should look for deleted submissions and incrememnt based of of that, but looking at
+    # production data I can't find a single case where that happens.
+    # For now ... I'm just gonna default it to 1?
```

Comment:
This comment is slightly too casual but if we don't come up with a better idea, I think leaving something like this in here is a good idea

Reply:
Also, to expand on this comment, what I did find in production was that there would be a deleted submission, and then an active submission with the same attempt_number